lib/registries/addon/overview/files/provider/route.ts

```diff
@@ -1,7 +1,33 @@
 import Route from '@ember/routing/route';
+import { waitFor } from '@ember/test-waiters';
+import { task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+
+import FileProviderModel from 'ember-osf-web/models/file-provider';
+import Registration from 'ember-osf-web/models/registration';
+import { GuidRouteModel } from 'ember-osf-web/resolve-guid/guid-route';
 
 export default class RegistrationFilesProviderRoute extends Route.extend({}) {
+    @task
+    @waitFor
+    async fileProviderTask(overview: GuidRouteModel<Registration>, fileProviderId: string) {
+        const registration = await overview.taskInstance;
+        const fileProviders = await registration!.files;
+        const provider = fileProviders.findBy('id', fileProviderId) as FileProviderModel;
```

Comment:
What happens if the file provider exists on the osf, but is not configured for this node? For example, on a registration if someone goes to `github/`?

Reply:
They will be rerouted to page-not-found. Registrations should only have osfstorage as a file-provider, so anything that isn't registrationId:osfstorage _should_ reroute after this